app/src/main/java/com/mobiledev/news_app/presentation/viewmodel/NewsViewModel.kt

```diff
@@ -20,13 +20,17 @@ class NewsViewModel(
     }
 
     fun onValueChange(transform: NewsState.() -> NewsState) {
-        _state.value = _state.value?.run(transform)
+        _state.value = _state.value?.transform()
     }
 
     fun fetchNews() {
         onValueChange {
             copy(
-                articles = repository.getNewsPage().cachedIn(viewModelScope)
+                articles = repository
+                    .getNewsPage(
+                        query = _state.value?.searchQuery,
```

Comment:
`searchQuery` is not part of view or ViewModel, it should be created in repository or use case level.